### PR TITLE
ci: use the native libcst parser

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

Use the native libcst parser. Its not yet compatiable with all 3.10 and 3.11 features (eg match statements and the new union format), but that's okay, as we're compatible with python 3.8 and so we don't use those features.

This should result in codemods being faster both locally and in ci.

~~Depending on when #958 is merged, a change will need to be made to add this environment to both codemods.~~

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `pdm lint`
    - [ ] I have type-checked the code by running `pdm pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by shiftinv** [comment]
Why not set this globally, similar to `PDM_IGNORE_SAVED_PYTHON`?

**Comment by onerandomusername** [comment]
> Why not set this globally, similar to `PDM_IGNORE_SAVED_PYTHON`?

somewhat updated in https://github.com/DisnakeDev/disnake/pull/969/commits/e77ebc4a54ebbbcb5583aaa16b38cbade7a4d7f6

File: noxfile.py [hunk 1]
```diff
@@ -102,6 +119,7 @@ def autotyping(session: nox.Session) -> None:
     """
     session.run_always("pdm", "install", "-dG", "codemod", external=True)
 
+    load_env_var("LIBCST_PARSER_TYPE", default="native")
```
Comment (shiftinv) [inline noxfile.py:122]: why not do this at the top level?

File: noxfile.py [hunk 2]
```diff
@@ -19,7 +19,24 @@
...
+    },
+)
+
+
+# load specific variables from the dotenv file if it exists
+def load_env_var(var: str, *, default: str, override: bool = False):
+    try:
+        import dotenv
+    except ImportError:
+        return default
```
Comment (shiftinv) [inline noxfile.py:34]: `os.environ` won't be set in this case, and the return value isn't being used anywhere.

Would it be easier to just do
```py
os.environ.setdefault(var, default)
```
at the top level and not do anything else? If someone wants to use their `.env`, `dotenv run` exists.

Comment (onerandomusername) [reply noxfile.py]: https://github.com/DisnakeDev/disnake/pull/969/commits/7986a840205b6fb7186b03a02cafd0aa0823a801

File: noxfile.py (hunk 1 above)
Comment (onerandomusername) [reply noxfile.py]: https://github.com/DisnakeDev/disnake/pull/969/commits/7986a840205b6fb7186b03a02cafd0aa0823a801

**Review by shiftinv — APPROVED** [review]

**Pull request merged**